Remotion/Core/Extensions/Logging/Log4Net/Log4NetLogger.cs

```diff
@@ -0,0 +1,139 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.Globalization;
+using log4net.Core;
+using log4net.Util;
+using MicrosoftLogging = Microsoft.Extensions.Logging;
+using RemotionLogging = Remotion.Logging;
+
+namespace Remotion.Logging.Log4Net;
+
+public class Log4NetLogger : MicrosoftLogging.ILogger
+{
+  private readonly log4net.Core.ILogger _logger;
+
+  public Log4NetLogger (string categoryName)
+  {
+    _logger = log4net.LogManager.GetLogger(categoryName).Logger;
+  }
+
+  public static Level Convert (MicrosoftLogging.LogLevel logLevel)
+  {
+    switch (logLevel)
+    {
+      case MicrosoftLogging.LogLevel.Debug:
+        return Level.Debug;
+      case MicrosoftLogging.LogLevel.Information:
+        return Level.Info;
+      case MicrosoftLogging.LogLevel.Warning:
+        return Level.Warn;
+      case MicrosoftLogging.LogLevel.Error:
+        return Level.Error;
+      case MicrosoftLogging.LogLevel.Critical:
+        return Level.Critical;
+      default:
+        throw new ArgumentException(string.Format("LogLevel does not support value {0}.", logLevel), "logLevel");
+    }
+  }
+
+  public static Level Convert (RemotionLogging.LogLevel logLevel)
+  {
+    switch (logLevel)
```

Comment:
We can use switch expressions here and in the other convert method, making it more cocise and IMO more readable: 
```
return logLevel switch 
{
 Loglevel.Debug => Level.Debug, 
.... 
}
```

Reply:
We should also support the rest of the `Microsoft.Extensions.LogLevels`, there should be appropriate Levels in log4net